### PR TITLE
node e2e conformance keep polling if encounters errors

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

**Comment by pwittrock** [comment]
@k8s test node e2e experimental

File: test/e2e_node/conformance_test.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ var _ = Describe("Container Conformance Test", func() {
 				phase := api.PodPending
 				for start := time.Now(); time.Since(start) < serviceCreateTimeout; time.Sleep(time.Second * 30) {
 					ccontainer, err := terminateCase.Get()
```
Comment (timstclair) [inline test/e2e_node/conformance_test.go]: Is ccontainer a typo, or intentional?

File: test/e2e_node/conformance_test.go [hunk 2]
```diff
@@ -62,7 +62,7 @@ var _ = Describe("Container Conformance Test", func() {
...
-					if err != nil || ccontainer.Phase != api.PodPending {
+					if err != nil && ccontainer.Phase != api.PodPending {
```
Comment (timstclair) [inline test/e2e_node/conformance_test.go]: Looking at https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/container.go#L93, won't the phase always be `""` when `err != nil`? It seems like maybe you just want to wait until `ccontainer.Phase != api.PodPending`?

File: test/e2e_node/conformance_test.go [hunk 3]
```diff
@@ -62,7 +62,7 @@ var _ = Describe("Container Conformance Test", func() {
 				phase := api.PodPending
```
Comment (timstclair) [inline test/e2e_node/conformance_test.go]: From [line 58](https://github.com/kubernetes/kubernetes/pull/23285/files#diff-803e7d016e077a89c74759ca80d447ceR58): It seems more like this is testing that the container stopped correctly?

File: test/e2e_node/conformance_test.go (hunk 1 above)
Comment (pwittrock) [inline test/e2e_node/conformance_test.go]: I thought it was intention, but now that you mention it I am not certain.

File: test/e2e_node/conformance_test.go (hunk 3 above)
Comment (pwittrock) [inline test/e2e_node/conformance_test.go]: It would probably be better to say it "it should run successfully"

Comment (timstclair) [inline test/e2e_node/conformance_test.go]: If that's the intention, I feel like you should add a sleep to the container run command, and verify it enters the `Running` phase.

Comment (pwittrock) [inline test/e2e_node/conformance_test.go]: Added a TODO and issue for this assigned to the origin author

File: test/e2e_node/conformance_test.go (hunk 2 above)
Comment (pwittrock) [inline test/e2e_node/conformance_test.go]: Done

**Comment by pwittrock** [comment]
PTAL

**Comment by pwittrock** [comment]
Rebased

**Comment by pwittrock** [comment]
@timstclair Thanks for your comments.  Let me know when you get a chance to take an other look.

File: test/e2e_node/conformance_test.go [hunk 4]
```diff
@@ -112,10 +112,11 @@ var _ = Describe("Container Conformance Test", func() {
 				err := terminateCase.Create()
 				Expect(err).NotTo(HaveOccurred())
 
+				// TODO: Check that the container enters running state by sleeping in the container #23309
 				phase := api.PodPending
 				for start := time.Now(); time.Since(start) < serviceCreateTimeout; time.Sleep(time.Second * 30) {
-					ccontainer, err := terminateCase.Get()
-					if err != nil || ccontainer.Phase != api.PodPending {
+					ccontainer, _ := terminateCase.Get()
+					if ccontainer.Phase != api.PodPending {
```
Comment (timstclair) [inline test/e2e_node/conformance_test.go]: Wouldn't it be better to poll until `container.Phase == terminateCase.Phase`? I could envision flakes from the container entering a waiting state, or being polled the moment it enter's running.

Comment (timstclair) [inline test/e2e_node/conformance_test.go]: Same below.

**Comment by pwittrock** [comment]
@k8s-bot test node e2e experimental

File: test/e2e_node/conformance_test.go (hunk 4 above)
Comment (pwittrock) [inline test/e2e_node/conformance_test.go]: Done.  I cleaned this up quite a bit using Eventually.

**Comment by pwittrock** [comment]
PTAL

**Comment by pwittrock** [comment]
@timstclair Mind taking another quick look?  Now that node e2e is a submit queue blocker, it is becoming more important to address flakiness to keep the merge queue happy.

**Comment by timstclair** [comment]
LGTM. Sorry I missed your first ping.

**Comment by pwittrock** [comment]
no worries :)  Thanks

**Comment by pwittrock** [comment]
@k8s-bot test this issue: #23610

**Comment by pwittrock** [comment]
@k8s-bot test node e2e experimental

**Pull request merged**